### PR TITLE
Support removing the actor, light, or model from the root

### PR DESCRIPTION
**Pull request by nkoenig**
# 🎉 New feature

## Summary

The Root object didn't have a mechanism to clear/remove the actor, light, or model. This PR adds a `Clear` function that similar to the other `Clear` functions.

## Test it
Tests have been added.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by ahcorde — APPROVED** [review]

File: src/Root_TEST.cc [hunk 1]
```diff
@@ -670,3 +670,33 @@ TEST(DOMRoot, WorldByName)
...
+{
+  sdf::Root root;
+  EXPECT_EQ(nullptr, root.Actor());
+  EXPECT_EQ(nullptr, root.Light());
+  EXPECT_EQ(nullptr, root.Model());
+
+  sdf::Actor actor1;
+  actor1.SetName("actor1");
+  root.SetActor(actor1);
+  EXPECT_NE(nullptr, root.Actor());
```
Comment (scpeters) [inline src/Root_TEST.cc:685]:
```suggestion
  EXPECT_NE(nullptr, root.Actor());
  EXPECT_EQ(nullptr, root.Light());
  EXPECT_EQ(nullptr, root.Model());
```
let's check all three pointers, expecting at most one to be non-null

File: src/Root_TEST.cc [hunk 2]
```diff
@@ -670,3 +670,33 @@ TEST(DOMRoot, WorldByName)
...
+  root.ClearActorLightModel();
+  EXPECT_EQ(nullptr, root.Actor());
```
Comment (scpeters) [inline src/Root_TEST.cc:687]:
```suggestion
  EXPECT_EQ(nullptr, root.Actor());
  EXPECT_EQ(nullptr, root.Light());
  EXPECT_EQ(nullptr, root.Model());
```

File: src/Root_TEST.cc [hunk 3]
```diff
@@ -670,3 +670,33 @@ TEST(DOMRoot, WorldByName)
...
+  EXPECT_NE(nullptr, root.Light());
+  root.ClearActorLightModel();
+  EXPECT_EQ(nullptr, root.Light());
+
+  sdf::Model model1;
+  model1.SetName("model1");
+  root.SetModel(model1);
+  EXPECT_NE(nullptr, root.Model());
+  root.ClearActorLightModel();
+  EXPECT_EQ(nullptr, root.Model());
```
Comment (scpeters) [inline src/Root_TEST.cc:701]:
```suggestion
  EXPECT_EQ(nullptr, root.Actor());
  EXPECT_EQ(nullptr, root.Light());
  EXPECT_EQ(nullptr, root.Model());
```

File: src/Root_TEST.cc [hunk 4]
```diff
@@ -670,3 +670,33 @@ TEST(DOMRoot, WorldByName)
...
+  EXPECT_NE(nullptr, root.Actor());
+  root.ClearActorLightModel();
+  EXPECT_EQ(nullptr, root.Actor());
+
+  sdf::Light light1;
+  light1.SetName("light1");
+  root.SetLight(light1);
+  EXPECT_NE(nullptr, root.Light());
+  root.ClearActorLightModel();
+  EXPECT_EQ(nullptr, root.Light());
```
Comment (scpeters) [inline src/Root_TEST.cc:694]:
```suggestion
  EXPECT_EQ(nullptr, root.Actor());
  EXPECT_EQ(nullptr, root.Light());
  EXPECT_EQ(nullptr, root.Model());
```

File: src/Root_TEST.cc [hunk 5]
```diff
@@ -670,3 +670,33 @@ TEST(DOMRoot, WorldByName)
...
+  actor1.SetName("actor1");
+  root.SetActor(actor1);
+  EXPECT_NE(nullptr, root.Actor());
+  root.ClearActorLightModel();
+  EXPECT_EQ(nullptr, root.Actor());
+
+  sdf::Light light1;
+  light1.SetName("light1");
+  root.SetLight(light1);
+  EXPECT_NE(nullptr, root.Light());
```
Comment (scpeters) [inline src/Root_TEST.cc:692]:
```suggestion
  EXPECT_EQ(nullptr, root.Actor());
  EXPECT_NE(nullptr, root.Light());
  EXPECT_EQ(nullptr, root.Model());
```

File: src/Root_TEST.cc [hunk 6]
```diff
@@ -670,3 +670,33 @@ TEST(DOMRoot, WorldByName)
...
+  root.ClearActorLightModel();
+  EXPECT_EQ(nullptr, root.Light());
+
+  sdf::Model model1;
+  model1.SetName("model1");
+  root.SetModel(model1);
+  EXPECT_NE(nullptr, root.Model());
```
Comment (scpeters) [inline src/Root_TEST.cc:699]:
```suggestion
  EXPECT_EQ(nullptr, root.Actor());
  EXPECT_EQ(nullptr, root.Light());
  EXPECT_NE(nullptr, root.Model());
```

**Review by scpeters — APPROVED** [review]
looks good; I suggested some small, optional improvements to the test

**Pull request merged**